src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -111,14 +113,19 @@ internal void Bind()
                     {
                         listenSocket.DualMode = true;
                     }
+
+                    ConfigureSocket();
```

Comment:
If we go the `Action<EndPoint, Socket>` route, wouldn't it be better to call it after binding but before listening? This would make the FileHandleEndPoint and IPEndPoint cases  more similar.

If we go the `Func<EndPoint, Socket>` route, I'd expect the returned Socket to already be bound.